### DISCUSSION TITLE
dhcpcd support for duid-en

### DISCUSSION THREAD

**Discussion by dbartolic** [rsmarples/dhcpcd#41] [Q&A]
Hi,

tried to configure dhcpcd to use duid-en (as defined in RFC3315), but without success. Browsing the source code shows that duid.h defines only duid-ll, duid-llt and duid-uuid (defined in RFC6355) but not duid-en (type 2), so it looks like duid-en is not supported at all.

Am I missing something or is using duid-en is not supported in dhcpcd?

Regards,
Damir Bartolic

**Comment by rsmarples** [comment]
So once a DUID is generated it is stored persistently on disk and constantly resused. All the current DUID options do is to provide a hint as to how this should be generated.

As dhcpcd has no idea how to generate one for your enterprise needs, it won't. However, nothing stops you from providing it on the DUID file as noted in the man page.

Does this work for you?
